gateway/ln-gateway/src/lib.rs

```diff
@@ -860,8 +834,13 @@ impl Gateway {
                     client.borrow().with_sync(|client| client.get_config_json()),
                 );
             } else {
-                let federation_clients = self.clients.read().await.clone().into_iter();
-                for (federation_id, client) in federation_clients {
+                for (federation_id, client) in self
```

Comment:
I think almost this entire function `handle_get_federation_config` should move into `FederationManager`

Reply:
Done. I split it into `get_federation_config()` and `get_all_federation_configs()` within `FederationManager`.